src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -346,17 +346,29 @@ def in_progress(self):
     @pass_app()
     async def get_ips(self, app):
         """
-        Get a list of IPs which can be accessed for management via UI.
+        Get a list of IPs for which the webUI can be accessed.
         """
-        addresses = (await self.middleware.call('system.general.config'))['ui_address']
-        if '0.0.0.0' in addresses:
-            ips = []
-            for interface in await self.middleware.call('interface.query', [
-                ('failover_vhid', '!=', None)
-            ]):
-                ips += [i['address'] for i in interface.get('failover_virtual_aliases', [])]
-            return ips
-        return addresses
+        v4addrs = (await self.middleware.call('system.general.config'))['ui_address']
+        v6addrs = (await self.middleware.call('system.general.config'))['ui_v6address']
+        all_ip4 = '0.0.0.0' in v4addrs
+        all_ip6 = '::' in v6addrs
+
+        if all_ip4 or all_ip6:
+            addrs = []
+            for i in await self.middleware.call('interface.query', [('failover_vhid', '!=', None)]):
+                # user can bind to a single v4 address but all v6 addresses
+                # or vice versa
+                addrs.extend([
+                    x['address'] for x in i.get('failover_virtual_aliases', [])
+                    if x['type'] == 'INET' and all_ip4
+                ])
+                addrs.extend([
+                    x['address'] for x in i.get('failover_virtual_aliases', [])
+                    if x['type'] == 'INET6' and all_ip6
+                ])
+            return addrs
```

Comment:
So if we have all ipv4 or all ipv6 we will only return these? and not selected ipv4 or ipv6?